src/darwin/Framework/CHIPTests/MTRXPCListenerSampleTests.m

```diff
@@ -501,25 +502,38 @@ - (BOOL)unitTestShouldSkipExpectedValuesForWrite:(MTRDevice *)device
 }
 @end
 
-@interface MTRXPCListenerSampleTests : XCTestCase
+@interface MTRXPCListenerSampleTests : MTRTestCase
 
 @end
 
 static BOOL sStackInitRan = NO;
-static BOOL sNeedsStackShutdown = YES;
 
 @implementation MTRXPCListenerSampleTests
 
++ (void)setUp
+{
+    // Global setup, runs once.
+    [super setUp];
+
+    __auto_type * uniqueName = [[NSUUID UUID] UUIDString];
+    __auto_type * uniquePath = [NSTemporaryDirectory() stringByAppendingPathComponent:uniqueName];
+    [[NSFileManager defaultManager] createFileAtPath:uniquePath
+                                            contents:[kSimpleLogContent dataUsingEncoding:NSUTF8StringEncoding]
+                                          attributes:nil];
+    BOOL started = [self startAppWithName:@"all-clusters"
+                                arguments:@[
+                                    @"--end_user_support_log",
+                                    uniquePath,
+                                ]
+                                  payload:kOnboardingPayload];
+    XCTAssertTrue(started);
+}
+
 + (void)tearDown
 {
     // Global teardown, runs once
-    if (sNeedsStackShutdown) {
-        // We don't need to worry about ResetCommissionee.  If we get here,
-        // we're running only one of our test methods (using
-        // -only-testing:MatterTests/MTROTAProviderTests/testMethodName), since
-        // we did not run test999_TearDown.
-        //        [self shutdownStack];
-    }
+    [self shutdownStack];
```

Comment:
should we delete the file here or we don't care and createFileAtPath will handle it if one exists?

Reply:
It's in a temp directory, so presumably it will be deleted whenever space is needed....

But yes, we could delete the file.  Let me add something for that.